bot/ctx.go

```diff
@@ -79,24 +79,41 @@ func DefaultArgsParser() ArgsParser {
 type Context struct {
 	*Subcommand
 	*state.State
-
-	// Descriptive (but optional) bot name
-	Name string
-
-	// Descriptive help body
-	Description string
-
+	// ErrorReplier is an optional function that allows changing how the error
+	// is replied. It overrides ReplyError and is only used for MessageCreate
+	// events.
+	//
+	// Note that errors that are passed in here will bypas FormatError; in other
+	// words, the implementation might only care about ErrorReplier and leave
+	// FormatError as it is.
+	ErrorReplier func(err error, src *gateway.MessageCreateEvent) api.SendMessageData
+	// ErrorLogger logs any error that anything makes and the library can't
+	// reply to the client. This includes any event callback errors that aren't
+	// Message Create.
+	ErrorLogger func(error)
 	// Called to parse message content, default to DefaultArgsParser().
 	ParseArgs ArgsParser
-
 	// Called to check a message's prefix. The default prefix is "!". Refer to
 	// NewPrefix().
 	HasPrefix Prefixer
-
-	// AllowBot makes the router also process MessageCreate events from bots.
-	// This is false by default and only applies to MessageCreate.
-	AllowBot bool
-
+	// FormatError formats any errors returned by anything, including the method
+	// commands or the reflect functions. This also includes invalid usage
+	// errors or unknown command errors. Returning an empty string means
+	// ignoring the error.
+	//
+	// By default, this field replaces all @ with @\u200b, which prevents an
+	// @everyone mention.
+	FormatError func(error) string
+	// Quick access map from event types to pointers. This map will never have
+	// MessageCreateEvent's type.
+	typeCache sync.Map // map[reflect.Type][]*CommandContext
+	// Descriptive help body
+	Description string
+	// Descriptive (but optional) bot name
```

Comment:
While not related to this change, this should be updated to follow the godoc format